### ISSUE TITLE
Google tab performs search for "Search" commands meant to open new result tabs

### ISSUE THREAD

**Issue by alexandra-martin** [mozilla/firefox-voice#1246]
### Prerequisites:
Mic permissions are enabled.
"Allow" or "Don't Allow" button from “Allow Firefox Voice to Collect Voice Transcriptions” pop-up needs to be clicked.
"A privacy reminder from Google" is reviewed and agreed and the yellow language pop-up is closed.
A Google tab is opened.

### STR:
1. Use the shortcut or click on the mic icon from the browser toolbar in a tab, other than Google.
2. Say or write a "Search" command meant to open a new tab, e.g. "What movies are playing at the cinema", "How to make fish tacos".

### Expected result:
A new tab opens with the result, without interfering with other opened tabs.

### Actual result:
The Google tab performs a search for the command made.

### Notes:
Reproduced on Mac 10.14 and Win10x64 with Firefox Nightly 76.0a1 (2020-03-11).
After performing the search, the Google tab sometimes closes and it's not shown in the hidden tabs list.
Making the command, from step 2, in the Google tab has the same result.

![11](https://user-images.githubusercontent.com/54401091/76505627-78806e00-6452-11ea-90d4-0be710b6a4fa.gif)

**Comment by thecreepaccess** [comment]
Assihn this bug to me

**Comment by thecreepaccess** [comment]
Assign this bug to me

**Comment by alexandra-martin** [comment]
@ianb or @awallin can help with issues assignment.

**Comment by shreyaa-s** [comment]
Hey @ianb can I work on this one?

**Comment by JM-Mendez** [comment]
For context, this was a regression I knowingly introduced in PR #1180 by not fully understanding the original functionality. The fix is localized and I know where it is if this needs to go out soon, since it broke existing functionality. But I wouldn't mind if someone else got a chance to work it.

**Comment by shreyaa-s** [comment]
When I tried to reproduce the issue, what I observed was that when i made a search the results opened in new tab. But if I were to again give a voice command for a search then the results would open in the same tab and not in a new one. Is this the issue, @ianb @alexandra-martin ?

**Comment by JM-Mendez** [comment]
@shreyaa-s yes that's the issue. It should always open a new search tab, while also remembering the search results for previously opened tabs.

**Comment by shreyaa-s** [comment]
Does it work this way: 
I search for something. Now I'm in a google page. I search for something else while in this tab, should it overwrite the previous search results with the new search results? Or should they open in a new tab?
If it is the second one, i.e. each result opens in a new tab, then what is the use of this part of code in `search.js` :
``` 
if (_searchTabId) {
    try {
      await browser.tabs.get(_searchTabId);
      return _searchTabId;
    } catch (e) {
      // Presumably the tab doesn't exist
      log.info("Error getting tab:", String(e));
      _searchTabId = null;
    }
  } 
```
Is it required in any other function that calls `openSearchTab()` ?

**Comment by JM-Mendez** [comment]
@shreyaa-s it should work the second way you mentioned: always open results in a new tab.

I called `openSearchTab()` because I erroneously based my changes on how the if branch `buildSettings.android` works, which calls `performSearch()`. 

Now I believe this is because of the way the firefox browser works in android, and is not intended to be used for desktop. I'm not sure if that's true, but I do know the bug is my use of `openSearchTab()`. Which will reuse an existing search tab if found.

BTW, I have more context on this intent, so if you have more questions you can reach me here or in the chatroom under `jmendez`. I'm happy to help you fix my mess 😄

**Comment by shreyaa-s** [comment]
@JM-Mendez should i refactor the function and pass a new tab every time? The `buildSettings.android` shows an internal error on desktop.

**Comment by alexandra-martin** [comment]
Verified on Mac 10.14 with Firefox Nightly 76.0a1 (2020-03-31).

![6](https://user-images.githubusercontent.com/54401091/78047937-a8e86780-7381-11ea-87f3-d5a9f3f85833.gif)

**Comment by alexandra-martin** [comment]
It seems this bug has resurfaced with current updates.
Reproduced on Mac 10.14 with Firefox Nightly 77.0a1 (2020-04-07).
Also for some reason the doorhanger changes size when the Google tab is opened. Maybe it wants to display a card, but there isn't one available in the Google tab?

![4](https://user-images.githubusercontent.com/54401091/78675889-5ca3a700-78ee-11ea-880c-8ebd2704799e.gif)

**Comment by Simpcyclassy** [comment]
I am unable to replicate this as well. Works as expected
